apps/text-extraction/app/aperturedb_io.py

```diff
@@ -0,0 +1,374 @@
+from schema import CrawlDocument, Segment, ImageBlock, FullTextBlock
+from typing import Iterator, Optional, Tuple
+from uuid import uuid4
+import math
+import json
+from datetime import datetime, timezone
+from aperturedb.CommonLibrary import create_connector, execute_query
+import logging
+from batcher import SymbolicBatcher
+
+logger = logging.getLogger(__name__)
+
+JOB_CLASS = "SegmentationJob"
+
+
+class AperturedbIO:
+    def __init__(self, crawl_id: str, batch_size: int = 100):
+        self.crawl_id = crawl_id
+        self.job_id = str(uuid4())
+        self.batch_size = batch_size
+        self.db = create_connector()
+        self.start_time = datetime.now(timezone.utc)
+        self.n_segments = 0
+        self.n_images = 0
+        self.n_full_texts = 0
+        self.batcher = SymbolicBatcher(
+            execute_query=self.execute_query,
+            batch_size=batch_size,
+            prolog=self._batcher_prolog)
+        self.current_document = None
+
+    def execute_query(self,
```

Comment:
using the same name is a little confusing to the reader, maybe just import CommonLibrary then you can call CommonLibrary.execute_query and it is obvious what is going on

Reply:
I could change the name. There are two purposes to abstracting this method:
* I want to flip the default on enforcing success.
* I was planning to add unit tests, so this is a mock-injection point

Reply:
yeah, I think overriding how execute_query works is smart, and given that you already have this interface making execute_query part of the interface makes sense.